geomstats/stratified_geometry/spider.py

```diff
@@ -0,0 +1,303 @@
+"""Class for the spider.
+
+Lead authors: Anna Calissano & Jonas Lueg
+"""
+import itertools
+
+import geomstats.backend as gs
+from geomstats.geometry.euclidean import EuclideanMetric
+from geomstats.stratified_geometry.stratified_spaces import (
+    Point,
+    PointSet,
+    PointSetGeometry,
+    _vectorize_point,
+)
+
+
+class SpiderPoint(Point):
+    r"""Class for points of the Spider.
+
+    Parameters
+    ----------
+    s : int
+        The stratum, an integer indicating the stratum the point lies in. If zero, then
+        the point is on the origin.
+    x : float
+        A positive number, the coordinate of the point. It must be zero if and only if
+        the stratum is zero, i.e. the origin.
+    """
+
+    def __init__(self, s, x):
+        super(SpiderPoint, self).__init__()
+        if s == 0 and x != 0:
+            raise ValueError("If the stratum is zero, x must be zero.")
+        self.s = s
+        self.x = x
+
+    def __repr__(self):
+        """Return a readable representation of the instance."""
+        return f"s{self.s}: {self.x}"
+
+    def __hash__(self):
+        """Return the hash of the instance."""
+        return hash((self.s, self.x))
+
+    def __eq__(self, other):
+        """Compare two points."""
+        return self.s == other.s and abs(self.x - other.x) < gs.atol
+
+    def to_array(self):
+        """Return the hash of the instance."""
+        return gs.array([self.s, self.x])
+
+
+class Spider(PointSet):
+    r"""Spider: a set of rays attached to the origin.
+
+    Parameters
+    ----------
+    rays : int
```

Comment:
For consistency with the codebase, I would suggest renaming this to n_rays. In the current codebase, variables that contain "number of" something, are prefixed with `n_`.